### PR TITLE
libgudev: 237 → 238

### PR DESCRIPTION
**Pull request by jtojnar**
###### Description of changes

cc @flokli for umockdev changes

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/umockdev/default.nix [hunk 1]
```diff
@@ -81,12 +84,25 @@ stdenv.mkDerivation rec {
     ln -s "$PWD/libumockdev-preload.so.0" "$out/lib/libumockdev-preload.so.0"
   '';
 
+  passthru = {
+    # libgudev is needed for an optional test but it itself relies on umockdev for testing.
```
Comment (flokli) [inline pkgs/development/libraries/umockdev/default.nix:88]: Hooh, interesting bootstrapping problem :-D But LGTM :+1:

**Review by flokli — APPROVED** [review]

**Comment by flokli** [comment]
Didn't check if it builds.

**Comment by Alanazane** [comment]
After Arch updated to 238 Steam is failing to launch for some users, this was found to not be a libgudev issue but a Steam one. Figured it might be worth noting here just in case Steam doesn't fix the issue before this is set to land.
However I was unable to reproduce the bug after rebuilding with this branch, so it may not effect users NixOS regardless :)

Steam issue for the bug here: https://github.com/ValveSoftware/steam-for-linux/issues/9805

**Comment by jtojnar** [comment]
Reading the [upstream issue](https://gitlab.gnome.org/GNOME/libgudev/-/issues/9), it is a clash between system and steam-runtime libgudev. Our `steam-run` does not appear to contain `libgudev` so I guess we are fine.

**Comment by trofi** [comment]
Bisect says commit eafa5c93d9cc7778207851f0929592c6039f72d1 `libgudev: 237 → 238` broke `upower` test:

```
$ nix build --no-link -f. -L upower
...
upower> Summary of Failures:
upower> 29/58 Tests.test_hidpp_touchpad                                  FAIL            0.49s   exit status 1
upower> 32/58 Tests.test_hidpp_touchpad_race                             FAIL            0.74s   exit status 1
```

**Comment by jtojnar** [comment]
Ugh, we noticed there were some flaky tests in https://github.com/NixOS/nixpkgs/pull/241586#issuecomment-1627384519 but did not bump it because it required libgudev 238. I guess it is safe to bump now, opened https://github.com/NixOS/nixpkgs/pull/243217

**Pull request merged**